students/BrianNisonger/session9/test_mailroom.py

```diff
@@ -0,0 +1,181 @@
+from mailroom import Donor
+from mailroom import Donors
+import statistics
+
+
+def test_make_donor():
+    donor = Donor("Fred Flinstone")
+    assert donor.name == "Fred Flinstone"
+
+
+def test_make_donor_with_donation():
+    donor = Donor("Fred Flinstone", [100, 200, 300, 400])
+    assert donor.name == "Fred Flinstone"
+    assert donor.donation == [100, 200, 300, 400]
+
+
+def test_add_donation():
+    donor = Donor("Fred Flinstone")
+    donor.add_donation(500)
+    assert donor.donation == [
+        500,
+    ]
+
+
+def test_append_donation():
+    donor = Donor("Fred Flinstone")
+    donor.add_donation(500, 365.00)
+    assert donor.donation == [500, 365.00]
+    donor.add_donation(1007)
+    assert donor.donation == [500, 365.00, 1007]
+
+
+def test_average():
+    donor = Donor("Fred Flinstone")
+    average = donor.make_average([500, 600, 800])
+    assert average == statistics.mean([500, 600, 800])
+
+
+def test_total():
+    donor = Donor("Fred Flinstone")
+    total = donor.make_total([100, 200, 300])
+    assert total == sum([100, 200, 300])
+
+
+def test_number_donations():
+    donor = Donor("Fred Flinstone")
+    number_of_donations = donor.num_donations([1, 2, 3, 4, 5])
+    assert number_of_donations == 5
+
+
+def test_thank_you():
+    donor = Donor("Fred Flinstone", [100, 200, 300])
+    thanks = donor.make_thank_you()
+    assert thanks == 'Dear Fred Flinstone, Thank you for your donation of $600.00. These funds help save the migratory butterflies of New South East.Thank you'
+
+
+def test_filename():
+    donor = Donor("Fred Flinstone")
+    filename = donor.make_filename()
+    assert filename == 'Fred_Flinstone.txt'
+
+
+def test_make_report_string():
+    donor = Donor("Fred Flinstone")
+    donation = [100, 200, 300, 400, 500]
+    report_line = donor.make_report_string(donation)
+    assert report_line == "Fred Flinstone       $1500.00         5               $300.00         "
+
+
+def test_make_Donors():
+    d1 = Donor("Fred Flinstone")
+    d2 = Donor("James Dean")
+    d3 = Donor("Jack the Ripper")
+    donor_list = [d1, d2, d3]
+    d_list = Donors(donor_list)
+    assert d_list.donor_list[0].name == "Fred Flinstone"
+
+
+def test_add_new_donor():
+    d1 = Donor("Fred Flinstone")
+    d2 = Donor("James Dean")
+    d3 = Donor("Jack the Ripper")
+    d_list = Donors([d1, d2, d3])
+    d4 = Donor("Mickey Mouse")
+    d_list.append(d4)
+    assert d_list.donor_list[3].name == "Mickey Mouse"
+    d_list.donor_list[3].add_donation(100)
+    assert d_list.donor_list[3].donation == [
+        100,
+    ]
+
+
+def test_make_check_Donors_attributes():
+    d1 = Donor("Fred Flinstone", [100, 200, 300, 400])
+    d2 = Donor("James Dean", [500, 600, 700, 800])
+    d3 = Donor("Jack the Ripper", [1, 2, 3, 4])
+    d_list = Donors([d1, d2, d3])
+    assert d_list.donor_list[1].donation == [500, 600, 700, 800]
```

Comment:
you don't want to be testing that it's the index-1 in the list -- find it by name, and then test.

That way, you could store it internally differently, and it would all still work.